docs/guides/hello.md

```diff
@@ -0,0 +1,1236 @@
+  <div id="cpp_generate">
+  <p>To generate the client and server side interfaces, run:</p>
+  <pre>
+  $ make helloworld.grpc.pb.cc helloworld.pb.cc
+  </pre>
+  <p>Which internally invokes the protocol buffer compiler as:</p>
+  <pre>
+  $ protoc -I ../../protos/ --grpc_out=. --plugin=protoc-gen-grpc=grpc_cpp_plugin ../../protos/helloworld.proto
+  $ protoc -I ../../protos/ --cpp_out=. ../../protos/helloworld.proto
+  </pre>
+  <p>This generates:</p>
+  <ul><li><code>helloworld.pb.h</code>, which declares classes for populating, serializing, and retrieving our <code>HelloRequest</code> and <code>HelloResponse</code> message types, and its implementation <code>helloworld.pb.cc</code>.</li>
+  <li><code>helloworld.grpc.pb.h</code>, which declares our generated service classes, and its implementation <code>helloworld.grpc.pb.cc</code>.</li>
+  </ul>
+  </div>
+  <div id="python_generate">
+  <p>To generate the client and server side interfaces:</p>
+  <pre>
+  $ ./run_codegen.sh
+  </pre>
+  <p>This invokes the protocol buffer compiler to generate <code>helloworld_pb2.py</code>, which contains our generated client and server classes, as well as classes for populating, serializing, and retrieving our <code>HelloRequest</code> and <code>HelloResponse</code> message types.</p>
+  </div>
+
+  <div id="go_generate">
+  <p>To generate the client and server side interfaces, run the protocol buffer compiler:</p>
+
+  <pre>
+  protoc -I ../protos ../protos/helloworld.proto --go_out=plugins=grpc:helloworld
+  </pre>
+
+  <p>This generates `helloworld.pb.go`, which contains our generated client and server code, as well as code for populating, serializing, and retrieving our `HelloRequest` and `HelloResponse` message types.</p>
+  </div>
+
+  <div id="ruby_generate">
+  <p>To generate the client and server side interfaces, run the protocol buffer compiler:</p>
+
+  <pre>
+  protoc -I ../protos --ruby_out=lib --grpc_out=lib --plugin=protoc-gen-grpc=`which grpc_ruby_plugin` ../protos/helloworld.proto
+  </pre>
+
+  <p>This generates the following files in the <code>lib</code> directory:</p>
+
+  <ul><li><code>lib/helloworld.rb</code> defines a module <code>Helloworld</code>, which provides all the protocol buffer code to populate, serialize, and retrieve our request and response message types.</li>
+  <li><code>lib/helloworld_services.rb</code> extends the <code>Helloworld</code> module with our generated client and server classes.</li>
+  </ul>
+  </div>
+
+  <div id="node_generate">
+  <p>The Node.js library dynamically generates service descriptors and client stub definitions from .proto files loaded at runtime, so there's no need to generate any special code when using this language. Instead, in our example server and client we <code>require</code> the gRPC library, then use its <code>load()</code> method:</p>
+
+  <pre>
+  var grpc = require('grpc');
+  var hello_proto = grpc.load(PROTO_PATH).helloworld;
+  </pre>
+  </div>
+
+  <div id="csharp_generate">
```

Comment:
@hsaliak, for csharp? Looks right to me, but we might want to note that they might have to change the permissions on the protobuf binaries in the Grpc.Tools package, in order to run them after restoring the package. 
The Grpc.Tools package in the examples I believe is currently at 0.15.0, but its at 0.14.0 here, but I think this will change to 1.0.0 probably anyways.
Also, we might want to note that they'll need to get the Grpc.Tools package themselves, with `nuget install Grpc.Tools`, if they're using the dotnet cli example (which is in PR right now https://github.com/grpc/grpc/pull/7634)